src/installer/managed/Microsoft.NET.HostModel/Bundle/Bundler.cs

```diff
@@ -273,80 +274,149 @@ public string GenerateBundle(IReadOnlyList<FileSpec> fileSpecs)
                 _tracer.Log($"Ovewriting existing File {bundlePath}");
             }
 
-            BinaryUtils.CopyFile(hostSource, bundlePath);
-
-            // Note: We're comparing file paths both on the OS we're running on as well as on the target OS for the app
-            // We can't really make assumptions about the file systems (even on Linux there can be case insensitive file systems
-            // and vice versa for Windows). So it's safer to do case sensitive comparison everywhere.
-            var relativePathToSpec = new Dictionary<string, FileSpec>(StringComparer.Ordinal);
-
             long headerOffset = 0;
-            using (FileStream bundle = File.Open(bundlePath, FileMode.Open, FileAccess.ReadWrite))
-            using (BinaryWriter writer = new BinaryWriter(bundle, Encoding.Default, leaveOpen: true))
+            string destinationDirectory = new FileInfo(bundlePath).Directory?.FullName ?? "";
+            if (!Directory.Exists(destinationDirectory))
             {
-                if (_target.IsOSX)
+                Directory.CreateDirectory(destinationDirectory);
+            }
+            using (FileStream bundleStream = File.Open(bundlePath, FileMode.Create, FileAccess.ReadWrite))
+            {
+                using (FileStream hostSourceStream = File.OpenRead(hostSource))
                 {
-                    MachObjectFile.RemoveCodeSignatureIfPresent(bundle);
+                    hostSourceStream.CopyTo(bundleStream);
```

Comment:
Is there any benefit to this being async? My guess is that MSBuild doesn't have an async entry point, but I could be wrong about that.